Add 01-intro (#1)
romankuchin/ica0002-2021: 01-intro/lab.md

Logout and log in again. Now this command should also work:

    ansible --version

## Task 4: Test access to your virtual machine

First, make sure that your Git repository is set up correctly -- check
[this list](http://193.40.156.86/students.html) for details.

Then, make sure that your virtual machine is set up -- click your name here and find the
SSH access details on your page. Note the SSH port number!

Finally, test that SSH access works -- run this command on the **control node**
(replace the `122` with the port number from the list above):

    ssh -p122 ubuntu@193.40.156.86 uptime

The command above should print the virtual machine uptime (a few minutes, maybe
hours). This means that you can access your virtual machine via SSH.

If you cannot access the virtual machine or it's stuck in 'Creating' state,
please ask the teachers for help.

## Task 5: Create Ansible playbook

**(on a Control node -- same machine you'll be running Ansible from)**

Example of first playbook you can find [here](01-demo).

Step 1: Create a working directory named `ica0002` (same as your GitHub
repository name).

Step 2: In that directory, create files named `ansible.cfg`, `hosts`,

> `roles/test_connection/main.yaml` and `infra.yaml` -- you can just copy

Reviewer: ```suggestion
`roles/test_connection/tasks/main.yaml` and `infra.yaml` -- you can just copy
```

# Lab 1

## Git Intro

You will keep all the code (and more) written during this course in a Git
repository.

Teachers will check your work in your Git repo -- not in your text editor.
Having all the task completed with results pushed to your Git repository is a
requirement to access the exam.

GitHub is a well-known and widely used (but not the only one) service to host
Git repositories. We'll use GitHub for this course as a collaboration platform.

If you feel some lack of experience with Git and GitHub we recommend to read
[this tutorial](https://guides.github.com/introduction/git-handbook).

> It's ok to use GUI version:[GitHub Desktop](https://desktop.github.com/)

Reviewer: ```suggestion
It's also ok to use GUI version if you prefer: [GitHub Desktop](https://desktop.github.com/)
```

version installed and use it to check your tasks. Your code is considered
working only if it executes successfully on Ansible 2.9. If you're using other
Ansible versions -- you're on your own with them.

Note for Windows users: Ansible does not officially support running on Windows,
neither do we support Windows as a workstation. Although we have nothing against
you get Ansible running on Windows, we recommend to install Ubuntu (Desktop
or Server) in VirtualBox, and run the Ansible from there -- it's simpler.

For Linux or OS X, we recommend to use Python virtual env:

    python3 -m venv ~/ansible-venv
    ~/ansible-venv/bin/pip install ansible==4.4.0
    ~/ansible-venv/bin/ansible --version

Last command should print something like

    ansible [core 2.11.4]

Then, add this line to your `~/.profile` file (if the file is missing, create it)
so you can use 'shorter' commands:

    PATH="$HOME/ansible-venv/bin:$PATH"

Logout and log in again. Now this command should also work:

    ansible --version

## Task 4: Test access to your virtual machine

First, make sure that your Git repository is set up correctly -- check
[this list](http://193.40.156.86/students.html) for details.

> Then, make sure that your virtual machine is set up -- SSH access details should
> be printed here: http://193.40.156.86/students/your_github_username.html. Note the SSH port number!

Reviewer: ```suggestion
Then, make sure that your virtual machine is set up -- click your name here and find the
SSH access details on your page. Note the SSH port number!
```

I'm afraid we'll spend a few hours explaining why http://193.40.156.86/students/your_github_username.html link does not work...

Finally, test that SSH access works -- run this command on the **control node**
(replace the `122` with the port number from the list above):

    ssh -p122 ubuntu@193.40.156.86 uptime

The command above should print the virtual machine uptime (a few minutes, maybe
hours). This means that you can access your virtual machine via SSH.

If you cannot access the virtual machine or it's stuck in 'Creating' state,
please ask the teachers for help.

## Task 5: Create Ansible playbook

**(on a Control node -- same machine you'll be running Ansible from)**

Example of first playbook you can find [here](01-demo).

Step 1: Create a working directory named `ica0002` (same as your GitHub
repository name).

Step 2: In that directory, create files named `ansible.cfg`, `hosts`,

> `roles/test_connection/main.yaml` and `test_ansible.yaml` -- you can just copy

Reviewer: ```suggestion
`roles/test_connection/tasks/main.yaml` and `test_ansible.yaml` -- you can just copy
```

the contents of `01-demo` directory from the link above.

Note: directory structure and file names matter! Create the files and
directories named exactly as requested.

> Step 3: Update your inventory file named `hosts` -- use
> http://193.40.156.86/students/your_github_username.html to find the correct connection
> parameters.

Reviewer: ```suggestion
Step 3: Update your inventory file named `hosts` -- check your own page from
http://193.40.156.86/students.html to find the correct connection parameters.
```

Step 4: Run the Ansible playbook:

>     ansible-playbook test_ansible.yaml

Reviewer: ```suggestion
    ansible-playbook infra.yaml
```

You should see only green "ok" messages.

## Task 6: Commit and your work to GitHub

Once all the previous tasks are done make sure the files you've created in the
Task 5 are pushed to GitHub.

Run this commands in the working directory (`ica0002` created in the Task 5) to
initialize the Git repository:

    git init
    git remote add origin <your-repository-url>

Exact commands can be found in GitHub, on your repository page.

Then, run these commands to commit your changes:

    git add .
    git commit -m 'Lab 1'

Finally, run this command to push your changes to GitHub:

    git push origin main

Once done, the following files should appear in the root of your GitHub
repository:
 - `ansible.cfg`
 - `hosts`

>  - `roles/test_connection/tasks/main.yaml`
>  - `test_ansible.yaml`

Reviewer: ```suggestion
 - `infra.yaml`
 - `roles/test_connection/tasks/main.yaml`
```
